actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -86,12 +101,16 @@ def etag(&etagger)
     #
     #   before_action { fresh_when @article, template: 'widgets/show' }
     #
-    def fresh_when(object = nil, etag: object, last_modified: nil, public: false, template: nil)
+    def fresh_when(object = nil, etag: nil, weak_etag: nil, strong_etag: nil, last_modified: nil, public: false, template: nil)
+      weak_etag ||= etag || object unless strong_etag
```

Comment:
Hello @jeremy ! I have a question about this change (if you remember… it's almost one year old).

Before this PR, I was able to use `fresh_when` to cache a page based *solely on last_modified*, and completely ignoring the body.

For instance, I could have an action like this:

```ruby
def index
  @sections = Section.all
  fresh_when @sections, etag: nil
end
```

which would render the page if any Section had been updated, or respond with 304 otherwise.

After this PR, I cannot find a way to not set an etag. 
Even if I pass `etag: nil` or `etag: false`, this line:

```ruby
weak_etag ||= etag || object unless strong_etag
```

ends up using `object` as the value of `etag`.

I understand the good intention of this PR: in principle, if your response has changed (and therefore Etag), you want to expire the cache. However, in my case, using `last_modified` is enough. If `last_modified` didn't change, then I want to serve the cached version without checking the Etag as well.

Am I correct in saying that this approach cannot be taken anymore after this PR?